### PR TITLE
add tests for arrays in generation of soap models

### LINKED ISSUE #123: Arc
How can I run loopback arc if you installed it with yeoman ? Documentation is missing this part. 

Create a model in your app
    $ yo loopback:model
  Compose your API, run, deploy, profile, and monitor it with Arc
    $ slc arc
  Run the app
    $ node .

### PR DESCRIPTION
**Pull request by emonddr**
Related to PR: https://github.com/strongloop/loopback-soap/pull/31 
Related to issue: https://github.com/strongloop/loopback-soap/issues/32

The tests for the generated API code are placed in `loopback-soap`.
The tests for the generated models are placed in `generator-loopback`.

My fix to issue https://github.com/strongloop/loopback-soap/issues/32 didn't involve arrays, but I was asked to make sure my fix in https://github.com/strongloop/loopback-soap/pull/31 could handle arrays.

Extra tests had to be added in this repository regarding models generated from WSDL files that have arrays in the operation input and/or output.

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/generator-loopback) 👈

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](https://loopback.io/doc/en/contrib/style-guide-es6.html)
- [ ] Commit messages are following our [guidelines](https://loopback.io/doc/en/contrib/git-commit-messages.html)

### CODE REVIEW

**Comment by rmg** [comment]
@slnode test please

**Comment by emonddr** [comment]
@slnode test please

File: package.json [hunk 1]
```diff
@@ -34,7 +34,7 @@
     "lodash": "^4.17.11",
     "loopback-api-definition": "^3.0.0",
     "loopback-bluemix": "^3.0.0",
-    "loopback-soap": "^1.0.0",
+    "loopback-soap": "github:strongloop/loopback-soap#dremond_fix_soap_gen",
```
Comment (dhmlau) [inline package.json:37]: please revert this change, or change the version accordingly.

File: test/soap/DataService_model_array.wsdl [hunk 2]
```diff
@@ -0,0 +1,61 @@
+<wsdl:definitions xmlns:apachesoap="http://xml.apache.org/xml-soap" xmlns:impl="http://stuff.dom" xmlns:intf="http://stuff.dom" xmlns:wsdl="http://schemas.xmlsoap.org/wsdl/" xmlns:wsdlsoap="http://schemas.xmlsoap.org/wsdl/soap/" xmlns:xsd="http://www.w3.org/2001/XMLSchema" targetNamespace="http://stuff.dom">
```
Comment (dhmlau) [inline test/soap/DataService_model_array.wsdl:1]: nitpick: is it possible to format the wsdl?

File: test/soap.test.js [hunk 3]
```diff
@@ -195,6 +195,134 @@ describe('loopback:soap tests', function() {
...
+            binding: 'DataServiceSoapBinding',
+            operations: ['getSomeData'],
+          }).then(function() {
+            var content = readModelJsonSync('get-some-data');
+            expect(content).to.not.have.property('public');
+            expect(content).to.have.property('properties');
+            expect(content.properties.myParams.type).to.be.an('array');
+            expect(content.properties.myParams.type[0]).to.eql('SomeInputData');
+
+            content = readModelJsonSync('some-data');
```
Comment (dhmlau) [inline test/soap.test.js:298]: just wondering... if the test is only for arrays, do we need to add the additional checks?

File: package.json (hunk 1 above)
Comment (emonddr) [reply package.json:37]: Yes, once I land the other PR, I will revert this. This PR is dependent on another PR.

File: test/soap.test.js (hunk 3 above)
Comment (emonddr) [reply test/soap.test.js:298]: @dhmlau ,there are previous tests which handle non-arrays and everything else they historically tested. Since my fix `adds` array-handling code, I've added tests for array situations. The other historical tests still handle non-array situations and ensure I haven't broken anything.

File: package.json (hunk 1 above)
Comment (rmg) [reply package.json:37]: If this PR depends on another PR then the other PR should be incrementing its version to `1.4.0`. After the other PR merges, this PR should be replacing the `github:...` specifier and incrementing the minimum version of the dependency to match the upstream change (`^1.4.0`).

File: test/soap.test.js [hunk 4]
```diff
@@ -53,7 +53,7 @@ describe('loopback:soap tests', function() {
             content = readModelJsonSync('globals');
             expect(content).to.not.have.property('public');
             expect(content).to.have.property('properties');
-            expect(content.properties.Promise.type).to.eql('boolean');
+            expect(content.properties.EventEmitter.type).to.eql('boolean');
```
Comment (deepakrkris) [inline test/soap.test.js:56]: @emonddr did the PR strongloop/loopback-soap#31 trigger this change in tests ? can you give some details ?

File: test/soap/DataService_model_array.wsdl (hunk 2 above)
Comment (emonddr) [reply test/soap/DataService_model_array.wsdl]: @dhmlau 

> nitpick: is it possible to format the wsdl? 


done  :)

File: test/soap.test.js (hunk 4 above)
Comment (emonddr) [reply test/soap.test.js:56]: @deepakrkris , sure. 

The historical test case was slightly wrong.

It was committing the same check twice:

```
content = readModelJsonSync('globals');
            expect(content).to.not.have.property('public');
            expect(content).to.have.property('properties');
            expect(content.properties.Promise.type).to.eql('boolean');
            expect(content.properties.Promise.type).to.eql('boolean');

```
It was checking that Promise.type is equal to boolean  `twice` .

The model it was checking was:

```
           <s:element name="GetQuote">
                <s:complexType>
                    <s:sequence>
                        <s:element minOccurs="0" maxOccurs="1" name="symbol" type="s:string"/>
                        <s:element name="globals" type="Globals"/>
                    </s:sequence>
                </s:complexType>
            </s:element>
            <s:complexType name="Globals">
                <s:sequence>
                    <s:element name="EventEmitter" type="s:boolean"/>
                    <s:element name="Promise" type="s:boolean"/>
                </s:sequence>
            </s:complexType>
```

it was forgetting to check

```
<s:element name="EventEmitter" type="s:boolean"/>
```

Comment (emonddr) [reply test/soap.test.js:56]: @deepakrkris , did I address your question?

Comment (rmg) [reply test/soap.test.js:56]: So this change isn't actually needed, it's more of a cleanup?

Might be better to do that in a separate PR or at least as a separate commit within this PR.

Comment (emonddr) [reply test/soap.test.js:56]: @rmg , I have placed the 'test case cleanup' in its own commit. :)

**Comment by emonddr** [comment]
@slnode test please

**Comment by emonddr** [comment]
@slnode test please
